source/brailleTables.py

```diff
@@ -636,3 +674,70 @@ def listTables():
 # Translators: The name of a braille table displayed in the
 # braille settings dialog.
 addTable("zu-za-g2.ctb", _("Zulu grade 2"), contracted=True)
+
+# Add new first maps for the custom tables - provided in the scratchpad directory
+# and/or by addons.
+_tables = _tables.new_child()
+_tablesDirs = _tablesDirs.new_child()
+
+
+def _loadTablesFromManifestSection(source: str, directory: str, tablesDict: dict):
+	for fileName, tableConfig in tablesDict.items():
+		addTable(
+			fileName=fileName,
+			displayName=tableConfig["displayName"],
+			contracted=tableConfig["contracted"],
+			input=tableConfig["input"],
+			output=tableConfig["output"],
+			source=source,
+		)
+
+
+def initialize():
+	# The builtin tables were added at import time to the parent map.
+	# Now, add the custom tables to the first map.
+	import addonHandler
+	for addon in addonHandler.getRunningAddons():
+		try:
+			tablesDict = addon.manifest.get("brailleTables")
+			if not tablesDict:
+				continue
+			log.debug(f"Found {len(tablesDict)} braille table entries in manifest for add-on {addon.name!r}")
+			directory = os.path.join(addon.path, "brailleTables")
+			if os.path.isdir(directory):
+				_tablesDirs[addon.name] = directory
+			_loadTablesFromManifestSection(addon.name, directory, tablesDict)
+		except Exception:
+			log.exception(f"Error while applying custom braille tables config from addon {addon.name!r}")
+
+	# Load the custom tables from the scratchpad last so it takes precedence over add-ons
+	if (
+		not globalVars.appArgs.secure
+		and config.conf['development']['enableScratchpadDir']

```

Comment:
consistency and to make code searching easier

```suggestion
		and config.conf["development"]["enableScratchpadDir"]
```